### PR TITLE
[Doc] Update documentation for Class was not found on autoload

### LINKED ISSUE #6841: Analyze error: Class was not found - a library class alias using class_alias()
# Question
Hello, I am using a PHP library which adds class aliases such as
```php
class_exists('hQuery_Element', false) or class_alias('duzun\\hQuery\\Element', 'hQuery_Element', false);
```
and whenever I run rector I get the following error
```php
[ERROR] Could not process "src/myFile.php" file, due to:                             
         "Analyze error: "Class duzun\hQuery\hQuery_Element was not found while trying to analyse it -    
         discovering symbols is probably not configured properly.". Include your files in                 
         "$parameters->set(Option::AUTOLOAD_PATHS, [...]);" or "$parameters->set(Option::BOOTSTRAP_FILES, 
         [...]);" in "rector.php" config.  
```
I am not sure what I should do is my situation. I am not using these aliases, but the library does and it breaks file analyzing.

### LINKED ISSUE #6852: Incorrect behavior of ReturnTypeDeclarationRector
# Bug Report

| Subject        | Details              |
| :------------- | :--------------------|
| Rector version | last dev-master      |
| Installed as   | composer dependency  |

## Minimal PHP Code Causing Issue

See https://getrector.org/demo/1ec58247-f7de-6f9e-8dd6-39bde39f72fc

```php
<?php

declare(strict_types=1);

/*
 * This file is part of the Serendipity HQ Stripe Bundle.
 *
 * Copyright (c) Adamo Aerendir Crespi <aerendir@serendipityhq.com>.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 */

namespace SerendipityHQ\Bundle\StripeBundle\Dev\Helper;

use Doctrine\ORM\Mapping\ClassMetadataInfo;
use Doctrine\ORM\Mapping\MappingException;
use SerendipityHQ\Bundle\StripeBundle\Dev\Doctrine\MappingFilesLocator;

class MappingHelper
{
    /** @var \Doctrine\ORM\Mapping\Driver\XmlDriver|null */
    private static $driver;
    /** @var mixed[] */
    private static $cache = [];

    public static function getMappedProperty(string $localModelClass, string $property): ?array
    {
        try {
            return self::getMetadataInfo($localModelClass)->getFieldMapping($property);
        } catch (MappingException $mappingException) {
        }

        return null;
    }

    private static function getMetadataInfo($localModelClass): ClassMetadataInfo
    {
        if (false === isset(self::$cache[$localModelClass])) {
            $metadataInfo = new ClassMetadataInfo($localModelClass);
            self::getDriver()->loadMetadataForClass($localModelClass, $metadataInfo);
            self::$cache[$localModelClass] = $metadataInfo;
        }

        return self::$cache[$localModelClass];
    }
}
```

### Responsible rules

* `ReturnTypeDeclarationRector`

## Expected Behavior

Rector shouldn't change anything.

The full class is here: https://github.com/Aerendir/bundle-stripe/blob/dev/dev/Helper/MappingHelper.php

### PR DESCRIPTION
**Pull request by samsonasik**
Based on possible tweak solution on:

- https://github.com/rectorphp/rector/issues/6841#issuecomment-991915633
- https://github.com/rectorphp/rector/issues/6852#issuecomment-991900246

by using "files" config in composer autoload as even can't be loaded by "classmap" config.

Fixes https://github.com/rectorphp/rector/issues/6841
Fixes https://github.com/rectorphp/rector/issues/6852

### CODE REVIEW

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

File: build/target-repository/docs/static_reflection_and_autoload.md [hunk 1]
```diff
@@ -90,3 +90,21 @@ In this case you may want to try one of the following solutions:
...
+        "files": [
+            "vendor/acme/my-custom-dependency/src/Your/Own/Namespace/TheAffectedClass.php"
+        ]
+    }
+```
+
+After that, run:
+
+```bash
+composer update
```
Comment (TomasVotruba) [inline build/target-repository/docs/static_reflection_and_autoload.md]: Composer would also update `compoesr.lock` and change dependencies.

Here the `composer dump-autoload`  would be enough

```suggestion
composer dump-autoload
```

Comment (samsonasik) [reply build/target-repository/docs/static_reflection_and_autoload.md]: updated https://github.com/rectorphp/rector-src/pull/1472/commits/66f744de903a273c76d4b6128ba71925b19bedb9

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

I updated the PR description that it fixes https://github.com/rectorphp/rector/issues/6841 and https://github.com/rectorphp/rector/issues/6852

**Comment by TomasVotruba** [comment]
Thank you :+1:

**Pull request merged**